expyfun/visual/_visual.py

```diff
@@ -1086,14 +1116,25 @@ def draw(self):
         self._player.update_texture()
         # detect end-of-stream to prevent pyglet from hanging:
         if not self._eos:
-            self._texture = self._player.get_texture()
-            self._scale_texture()
-            self._texture.blit(*self._actual_pos)
+            if self._visible:
+                self._texture = self._player.get_texture()
+                self._scale_texture()
+                self._texture.blit(*self._actual_pos)
         else:
             self._finished = True
             self.pause()
         self._ec.check_force_quit()
 
+    def set_visible(self, show, flip=False):
```

Comment:
The idea of having the flip arg is to remind users that it won't take effect until the next flip. The idea of defaulting to false is to not do things like flips automatically without very strong reasons. That was my thinking anyway, but I'm happy to defer if you think it is silly or confusing.